plone/app/upgrade/v50/final.py

```diff
@@ -39,3 +40,20 @@ def _migrate_list(original_id, new_id=None):
 def to501(context):
     """5.0 -> 5.0.1"""
     loadMigrationProfile(context, 'profile-plone.app.upgrade.v50:to501')
+
+    def fix_get_icon(context): 
+        """
+        get_icon redefined: now boolean: 
+        true if dexterity item is image or has image field (named 'image') e.g. leadimage 
+        see https://github.com/plone/Products.CMFPlone/issues/1226
+        """ 
+        catalog = getToolByName(context, 'portal_catalog')
+        search = catalog.unrestrictedSearchResults
+        cnt=0
+        iface = "plone.dexterity.interfaces.IDexterityContent"
+        for brain in search(object_provides=iface):
+            brain._unrestrictedGetObject().reindexObject()
```

Comment:
Indexing all indexes is not necessary and may take a long time. Indexing one index ix much faster.
I propose to only reindex the specific index like so::

    brain._unrestrictedGetObject().reindexObject(idx=['getIcon'])

I double checked if metadata gets updated if ``idx`` kwarg is given: This calls in depth then https://github.com/zopefoundation/Products.ZCatalog/blob/2.13.27/src/Products/ZCatalog/Catalog.py#L309 where all metadata is updated even if one index is given.